src/commands/react.ts

```diff
@@ -0,0 +1,168 @@
+
+import * as Discord from 'discord.js';
+
+import { Inject, AutoWired, Singleton } from 'typescript-ioc';
+
+import { ICommand, ICommandArgs, ICommandResult } from '../interfaces';
+import { EmojiService } from '../services/EmojiService';
+import { Logger } from '../services/Logger';
+import { ReactService } from '../services/ReactService';
+import { ErrorMessageService } from '../services/ErrorMessageService';
+import { Channel, DMChannel, GroupDMChannel, Message, TextChannel } from 'discord.js';
+
+@Singleton
+@AutoWired
+export class ReactCommand implements ICommand {
+
+  help = 'React to the message immediately above the command or react to a specific' +
+    'message with a message-id\n' +
+    '**Usage**\n' +
+    'react message-id emoji-name - reacts to a specific message\n' +
+    'react emoji-name - reacts to the above message\n' +
+    '**Supported Servers**\n' +
+    'PMMM: Magia Record\n' +
+    '/r/MadokaMagica\n' +
+    'Madoka\'s Disciples\n' +
+    'Magireco Emotes for Discord Whales\n' +
+    'and other partnered Emoji servers';
+  aliases = ['react', 'r'];
+
+  @Inject private logger: Logger;
+  @Inject private emojiService: EmojiService;
+  @Inject private reactService: ReactService;
+  @Inject private errorMessageService: ErrorMessageService;
+
+  async execute(cmdArgs: ICommandArgs): Promise<ICommandResult> {
+    const {message, args} = cmdArgs;
+
+    const commandArgs = args.split(' ');
+
+    // Args should only be length 1 or 2
+    if (commandArgs.length !== 1 && commandArgs.length !== 2) {
+      // Error, number of arguments doesn't match
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  You must either include an emoji name or a message-id and then an emoji name!');
+      if (message.deletable) {
```

Comment:
Maybe add a parameter to `sendErrorMessage` that does this automatically.

Reply:
Sure, a parameter there sounds good

Reply:
- [ ] Maybe add a parameter to sendErrorMessage that does this automatically.